### PR TITLE
timeout support for guild members

### PR DESCRIPTION
**Pull request by Wolfhound905**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition 

## Description

Added a function to timeout a member and added communication_disabled_until to the Member class
Example of features:
```py
@slash_command("timeout", "mute someone", scopes=[780435741650059264])
@slash_option("member", "person", opt_type=OptionTypes.USER)
@slash_option("unix_timestamp", "unixtimestamp.com", opt_type=OptionTypes.INTEGER)
async def mute(ctx: InteractionContext, member: Member, unix_timestamp: int):
    await member.timeout(unix_timestamp)
    await ctx.send(
        f"{member.mention} will be able to chat again {member.communication_disabled_until.format(TimestampStyles.RelativeTime)}"
    )

```

## Changes

- Timeout support

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord_objects/user.py [hunk 1]
```diff
@@ -423,6 +431,28 @@ def has_role(self, *roles: Union[Snowflake_Type, Role]) -> bool:
...
 
+    async def timeout(self, communication_disabled_until: Union[Timestamp, datetime] = MISSING, reason: str = MISSING):
+        """
+        Disable a members communication for a given time.
+
+        Args:
+            communication_disabled_until: The time until the user can communicate again (default 1 minute)
+        """
+        if communication_disabled_until is MISSING:
+            communication_disabled_until = Timestamp.now() + timedelta(minutes=1)
```
Comment (LordOfPolls) [inline dis_snek/models/discord_objects/user.py]: This is undocumented and inherantly unpredictable behaviour. This method should not have a default duration for timing someone out, it should be explictly set.

File: dis_snek/models/discord_objects/user.py [hunk 2]
```diff
@@ -423,6 +431,28 @@ def has_role(self, *roles: Union[Snowflake_Type, Role]) -> bool:
                 return False
         return True
 
+    async def timeout(self, communication_disabled_until: Union[Timestamp, datetime] = MISSING, reason: str = MISSING):
+        """
+        Disable a members communication for a given time.
+
+        Args:
+            communication_disabled_until: The time until the user can communicate again (default 1 minute)
```
Comment (LordOfPolls) [inline dis_snek/models/discord_objects/user.py]: reason is not in the docstring

**Review by LordOfPolls — CHANGES_REQUESTED** [review]

**Comment by Wolfhound905** [comment]
Ok, that was the last commit lol. I meant to rename a commit that I misspelled, but I wound up merging and stuff. Oops.
Nvm, one more because there are a few missing perms

File: dis_snek/models/enums.py [hunk 3]
```diff
@@ -318,6 +318,9 @@ class Permissions(DiscordIntFlag):  # type: ignore
     USE_PUBLIC_THREADS = 1 << 35
     USE_PRIVATE_THREADS = 1 << 36
     USE_EXTERNAL_STICKERS = 1 << 37
+    SEND_MESSAGES_IN_THREADS = 1 << 38
+    START_EMBEDDED_ACTIVITIES = 1 << 39
+    MODERATE_MEMBERS = 1 << 40
```
Comment (LordOfPolls) [inline dis_snek/models/enums.py:323]: This would have been merged, until you did this

**Review by LordOfPolls — APPROVED** [review]

**Comment by philip-ulrich** [comment]
🎉 Excited for this. Thanks @Wolfhound905

**Pull request merged**